core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -1010,6 +1011,82 @@ private FileIO tableFileIO(
     return newFileIO(context, fullConf, storageCredentials);
   }
 
+  /**
+   * Create a new {@link RESTTableOperations} instance for simple table operations.
+   *
+   * <p>This method can be overridden in subclasses to provide custom table operations
+   * implementations.
+   *
+   * @param restClient the REST client to use for communicating with the catalog server
+   * @param path the REST path for the table
+   * @param headers a supplier for additional HTTP headers to include in requests
+   * @param fileIO the FileIO implementation for reading and writing table metadata and data files
+   * @param current the current table metadata
+   * @param supportedEndpoints the set of supported REST endpoints
+   * @return a new RESTTableOperations instance
+   */
+  protected RESTTableOperations newTableOps(
```

Comment:
I'm just curious here: These are protected, so not part of any API or so. This means that there are no guarantees the signature of these functions don't change and any change here is not considered a breaking change. I don't think revAPI checks would break if I add a couple of extra parameters and change the types for some of the existing ones.
However, if you make your application so that it overrides these functions it could break with every Iceberg version bump. Isn't this an issue at your end?

Reply:
If someone decides to override this method, they're responsible for adapting to future changes. The Iceberg spec is continuously evolving, even if we define an API today, it doesn't mean it will remain unchanged.
For example, there's another ongoing PR that adds encryption-related properties to `RESTTableOperations`. So it’s quite common for APIs in this area to evolve over time.

But the benefits of just injecting a custom table operations is: if the new changes are not related to this method, we can upgrade smoothly.